src/cluster.c

```diff
@@ -108,6 +108,27 @@ ConnectionType *connTypeOfCluster(void) {
     return connectionTypeTcp();
 }
 
+void delUnownedKeys(void) {
+    if (!server.cluster_enabled) {
+        return;
+    }
+    clusterNode *primary;
+    if (clusterNodeIsPrimary(server.cluster->myself)) {
+        primary = server.cluster->myself;
+    } else {
+        primary = server.cluster->myself->replicaof;
+    }
+    for (unsigned int i = 0; i < CLUSTER_SLOTS; i++) {
+        if (server.cluster->slots[i] != primary) {
+            unsigned int deleted = delKeysInSlot(i);
+            if (deleted > 0) {
+                serverLog(LL_NOTICE, "Deleted %u keys from unowned slot: %d after loading RDB", deleted, i);
```

Comment:
Can we make it `LL_VERBOSE`, otherwise in the worst case, this might dump lot's of log line (one per missing slot (16384)).